Automattic/wp-calypso
# Theme Tiers: Design Picker aware of Personal Themes

Related to:
- https://github.com/Automattic/dotcom-forge/issues/5155

## Proposed Changes

* Added the necessary logic to the Design Picker so that it can identify themes on the Personal tier.
* When a Personal tier theme is selected, the `Unlock` button is shown.
* When clicking on the `Unlock` button the `Upgrade Modal` will be displayed.
* When clicking the `Upgrade Modal's` CTA the user is redirected to Checkout with the Personal plan selected.

## Testing Instructions



* Use the Calypso Live link in this PR.
* Go to the Design Picker and pick a Personal Theme
* You should see that the theme is gated.
* When clicking on the CTA the upgrade modal should be shown
* When clicking on the modal's CTA, you should be re-directed to the checkout flow with the Personal/Starter Plan selected by default.

Before:
![Screenshot 2024-01-15 at 15 15 19](https://github.com/Automattic/wp-calypso/assets/1989914/99389665-3d29-4384-bdad-50f17fe2e0a2)

After:
![Screenshot 2024-01-15 at 15 15 49](https://github.com/Automattic/wp-calypso/assets/1989914/e9a1c60c-8e46-4d3c-be17-705b0e654970)
![Screenshot 2024-01-15 at 15 15 56](https://github.com/Automattic/wp-calypso/assets/1989914/98dbf294-631a-43fb-91e0-e418f2856016)
![Screenshot 2024-01-15 at 15 16 13](https://github.com/Automattic/wp-calypso/assets/1989914/0d571f70-24a4-4b66-b025-f0f19b6c764c)


## Pre-merge Checklist
